### PR TITLE
Update Hugo version

### LINKED ISSUE #33001: Error on running the website locally using newer Hugo version
**This is a Bug Report**

**Problem:**

I followed the section "Running the website locally using Hugo" in README of this repo, but I met an error.

With
- Ubuntu 18.04 desktop amd64
- the latest commit of this repo (0ea4666)
- Docsy whose version is the same as of https://github.com/kubernetes/website/tree/main/themes (docsy @ 1c77bb2)
- npm version: 6.14.15
- node version: v14.18.1
- go version go1.17.3 linux/amd64
- hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended linux/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=mage
- Docker Client/Server: 20.10.11

---

[How to reproduce]

```
❯ git submodule update --init --recursive --depth 1
# OK with no output
```

```
❯ npm ci
npm WARN prepare removing existing node_modules/ before installation

> core-js-pure@3.6.5 postinstall /home/jhseo/k8s-website/node_modules/core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

added 120 packages in 1.414s
```

```
❯ make serve
hugo server --buildFuture --environment development
Start building sites … 
hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended linux/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=mage
WARN 2022/04/18 15:43:16 The google_news internal template will be removed in a future release. Please remove calls to this template. See https://github.com/gohugoio/hugo/issues/9172 for additional information.
WARN 2022/04/18 15:43:17 .Path when the page is backed by a file is deprecated and will be removed in a future release. We plan to use Path for a canonical source path and you probably want to check the source is a file. To get the current behaviour, you can use a construct similar to the one below:

  {{ $path := "" }}
  {{ with .File }}
        {{ $path = .Path }}
  {{ else }}
        {{ $path = .Path }}
  {{ end }}


Re-run Hugo with the flag --panicOnWarning to get a better error message.
Error: Error building site: "/home/jhseo/k8s-website/content/en/docs/concepts/workloads/controllers/statefulset.md:410:12": failed to render shortcode "api-reference": failed to process shortcode: "/home/jhseo/k8s-website/layouts/shortcodes/api-reference.html:7:17": execute of template failed: template: shortcodes/api-reference.html:7:17: executing "shortcodes/api-reference.html" at <$page.URL>: can't evaluate field URL in type page.Page
Built in 2752 ms
Makefile:54: recipe for target 'serve' failed
make: *** [serve] Error 255
```

**Proposed Solution:**

### PR DESCRIPTION
**Pull request by jihoon-seo**
Relevant with #33001.
We need to check whether the Netlify preview of this PR works well with no problem.

### CODE REVIEW

**Comment by sftim** [comment]
@jihoon-seo have you verified that this works OK with a local container image preview?

**Comment by sftim** [comment]
/retitle Update Hugo version

The version we configure for Netlify is also used for local previews in a container.

**Comment by tengqm** [comment]
I believe the change to the `api-reference.md` file is the only complaint from running the new Hugo version.
`0.87.0` is history. Hugo doesn't have a `release-0.87` branch now because it is "too" old.

File: layouts/shortcodes/api-reference.html [hunk 1]
```diff
@@ -4,4 +4,4 @@
 {{ $textArg := .Get "text" }}
 {{ $page := site.GetPage "page" (printf "%s/%s" $base $pageArg) }}
 {{ $metadata := $page.Params.api_metadata }}
-<a href="{{ $page.URL }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>
\ No newline at end of file
+<a href="{{ $page.RelPermalink }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>
```
Comment (sftim) [inline layouts/shortcodes/api-reference.html:7]: I'd separate this into its own commit, or even into its own PR (the change to this line is a good fix and easy to accept).

**Comment by jihoon-seo** [comment]
/hold
I am working on https://github.com/kubernetes/website/pull/33048#issuecomment-1103786125.

**Comment by jihoon-seo** [comment]
@sftim As per your comment, I
- separated into 2 PRs
  - #33073
  - #33048 (This one)
- verified that this works OK with a local container image preview, but only with updating the base Golang image as well

Keep hold for #33073 to be merged.
After #33073 gets merged, I will rebase this PR to latest `main`, then Netlify build of this PR will succeed.

@sftim @tengqm PTAL!

---

[Details]

With
- `netlify.toml`
```diff
-HUGO_VERSION = "0.87.0"
+HUGO_VERSION = "0.97.0"
```
- `Dockerfile`
```Dockerfile
FROM golang:1.16-alpine
```

[Output of building container image]
: Error occurs when building Hugo (extended) v0.97.0 binary in `golang:1.16-alpine` as a base image.
```
❯ make container-image           
docker build . \
        --network=host \
        --tag gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.97.0-c8ffb2b5979c \
        --build-arg HUGO_VERSION=0.97.0
Sending build context to Docker daemon  4.096kB
Step 1/12 : FROM golang:1.16-alpine
 ---> bef5ede0cd8a
Step 2/12 : LABEL maintainer="Luc Perkins <lperkins@linuxfoundation.org>"
 ---> Using cache
 ---> ea1b9510c298
Step 3/12 : RUN apk add --no-cache     curl     gcc     g++     musl-dev     build-base     libc6-compat
 ---> Using cache
 ---> f05caa3e7415
Step 4/12 : ARG HUGO_VERSION
 ---> Using cache
 ---> 2fcb154e3840
Step 5/12 : RUN mkdir $HOME/src &&     cd $HOME/src &&     curl -L https://github.com/gohugoio/hugo/archive/refs/tags/v${HUGO_VERSION}.tar.gz | tar -xz &&     cd "hugo-${HUGO_VERSION}" &&     go install --tags extended
 ---> Running in 932a1a60226b
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   132  100   132    0     0    443      0 --:--:-- --:--:-- --:--:--   444
100 26.5M    0 26.5M    0     0  5688k      0 --:--:--  0:00:04 --:--:-- 6772k
go: golang.org/x/tools@v0.1.10 requires
        github.com/yuin/goldmark@v1.4.1: missing go.sum entry; to add it:
        go mod download github.com/yuin/goldmark
The command '/bin/sh -c mkdir $HOME/src &&     cd $HOME/src &&     curl -L https://github.com/gohugoio/hugo/archive/refs/tags/v${HUGO_VERSION}.tar.gz | tar -xz &&     cd "hugo-${HUGO_VERSION}" &&     go install --tags extended' returned a non-zero code: 1
Makefile:69: recipe for target 'container-image' failed
make: *** [container-image] Error 1
```

---

With
- `netlify.toml`
```diff
-HUGO_VERSION = "0.87.0"
+HUGO_VERSION = "0.97.0"
```
- `Dockerfile`
```diff
-FROM golang:1.16-alpine
+FROM golang:1.18-alpine
```

[Output of building container image]
```
❯ make container-image
docker build . \
        --network=host \
        --tag gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.97.0-f2d1da00ba59 \
        --build-arg HUGO_VERSION=0.97.0
Sending build context to Docker daemon  4.096kB
Step 1/12 : FROM golang:1.18-alpine
 ---> dd6fd110e957
Step 2/12 : LABEL maintainer="Luc Perkins <lperkins@linuxfoundation.org>"
 ---> Using cache
 ---> 421386b052bb
Step 3/12 : RUN apk add --no-cache     curl     gcc     g++     musl-dev     build-base     libc6-compat
 ---> Using cache
 ---> a9bf7104f4f0
Step 4/12 : ARG HUGO_VERSION
 ---> Using cache
 ---> 28a5b14790be
Step 5/12 : RUN mkdir $HOME/src &&     cd $HOME/src &&     curl -L https://github.com/gohugoio/hugo/archive/refs/tags/v${HUGO_VERSION}.tar.gz | tar -xz &&     cd "hugo-${HUGO_VERSION}" &&     go install --tags extended
 ---> Using cache
 ---> 108739172956
Step 6/12 : FROM golang:1.18-alpine
 ---> dd6fd110e957
Step 7/12 : RUN apk add --no-cache     runuser     git     openssh-client     rsync     npm &&     npm install -D autoprefixer postcss-cli
 ---> Using cache
 ---> 850affda233d
Step 8/12 : RUN mkdir -p /var/hugo &&     addgroup -Sg 1000 hugo &&     adduser -Sg hugo -u 1000 -h /var/hugo hugo &&     chown -R hugo: /var/hugo &&     runuser -u hugo -- git config --global --add safe.directory /src
 ---> Using cache
 ---> 96799c6b8245
Step 9/12 : COPY --from=0 /go/bin/hugo /usr/local/bin/hugo
 ---> Using cache
 ---> 4261dee09e4e
Step 10/12 : WORKDIR /src
 ---> Using cache
 ---> 0f5ffdede52e
Step 11/12 : USER hugo:hugo
 ---> Using cache
 ---> 52217bb059e2
Step 12/12 : EXPOSE 1313
 ---> Using cache
 ---> c9df62533772
Successfully built c9df62533772
Successfully tagged gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.97.0-f2d1da00ba59
```

[Output of running container image]
```
❯ make container-serve
docker run --rm --interactive --tty --volume /home/jhseo/k8s-website:/src --cap-drop=ALL --cap-add=AUDIT_WRITE --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 gcr.io/k8s-staging-sig-docs/k8s-website-hugo:v0.97.0-f2d1da00ba59 hugo server --buildFuture --environment development --bind 0.0.0.0 --destination /tmp/hugo --cleanDestinationDir
Start building sites … 
hugo v0.97.0+extended linux/amd64 BuildDate=unknown
WARN 2022/04/21 01:46:24 The google_news internal template will be removed in a future release. Please remove calls to this template. See https://github.com/gohugoio/hugo/issues/9172 for additional information.
WARN 2022/04/21 01:46:25 .Path when the page is backed by a file is deprecated and will be removed in a future release. We plan to use Path for a canonical source path and you probably want to check the source is a file. To get the current behaviour, you can use a construct similar to the one below:

  {{ $path := "" }}
  {{ with .File }}
        {{ $path = .Path }}
  {{ else }}
        {{ $path = .Path }}
  {{ end }}


Re-run Hugo with the flag --panicOnWarning to get a better error message.
Error: Error building site: "/src/content/en/docs/concepts/workloads/controllers/statefulset.md:410:12": failed to render shortcode "api-reference": failed to process shortcode: "/src/layouts/shortcodes/api-reference.html:7:17": execute of template failed: template: shortcodes/api-reference.html:7:17: executing "shortcodes/api-reference.html" at <$page.URL>: can't evaluate field URL in type page.Page
Built in 3697 ms
Makefile:78: recipe for target 'container-serve' failed
make: *** [container-serve] Error 255
```
(This error will be resolved when #33073 gets merged.)

---

If I update `layouts/shortcodes/api-reference.html` as per #33073:
```diff
-<a href="{{ $page.URL }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>
+<a href="{{ $page.RelPermalink }}{{if $anchorArg}}#{{ $anchorArg }}{{end}}">{{if $textArg}}{{ $textArg }}{{else if $anchorArg}}{{ $anchorArg }}{{else}}{{ $metadata.kind }}{{end}}</a>
```

`make container-serve` runs well without error.
```
...

                   |  EN  |  ZH  | KO  | JA  | FR  | IT  | DE  | ES  | PT-BR | ID  | RU  | VI  | PL  | UK   
-------------------+------+------+-----+-----+-----+-----+-----+-----+-------+-----+-----+-----+-----+------
  Pages            | 1439 | 1004 | 539 | 448 | 338 |  71 | 164 | 292 |   183 | 335 | 155 |  77 |  69 |  92  
  Paginator pages  |   42 |    8 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  
  Non-page files   |  507 |  367 | 200 | 266 |  73 |  20 |  17 |  33 |    30 | 105 |  24 |   8 |   6 |  20  
  Static files     |  834 |  834 | 834 | 834 | 834 | 834 | 834 | 834 |   834 | 834 | 834 | 834 | 834 | 834  
  Processed images |    1 |    1 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  
  Aliases          |    8 |    1 |   3 |   1 |   0 |   1 |   0 |   0 |     1 |   1 |   1 |   0 |   0 |   0  
  Sitemaps         |    2 |    1 |   1 |   1 |   1 |   1 |   1 |   1 |     1 |   1 |   1 |   1 |   1 |   1  
  Cleaned          |    0 |    0 |   0 |   0 |   0 |   0 |   0 |   0 |     0 |   0 |   0 |   0 |   0 |   0  

Built in 10959 ms
Watching for changes in /src/{archetypes,assets,content,data,i18n,layouts,package.json,postcss.config.js,static,themes}
Watching for config changes in /src/config.toml, /src/themes/docsy/config.toml, /src/go.mod
Environment: "development"
Serving pages from /tmp/hugo
Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
Web Server is available at http://localhost:1313/ (bind address 0.0.0.0)
Press Ctrl+C to stop
```

**Comment by jihoon-seo** [comment]
I rebased this PR to latest `main`, and confirmed that `make container-image` && `make container-serve` succeeds with no problem.
/unhold

**Comment by zacharysarah** [comment]
@jihoon-seo Thanks for this. ✨ 

It looks like concern about compatibility with the API reference was addressed and merged successfully in #33073.

/lgtm
